crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -630,6 +630,81 @@ impl IndexerReader {
         stored_txn.try_into_iota_transaction_effects()
     }
 
+    /// Expensive check to assert whether all transactions
+    /// are indexed.
+    ///
+    /// It uses both the `tx_global_order` table
+    /// and the `checkpoints` table to cover old transactions.
+    fn deep_check_all_transactions_are_indexed(
+        &self,
+        digests: &[TransactionDigest],
+    ) -> Result<bool, IndexerError> {
+        let (checkpointed, optimistic) = self
+            .multi_get_transactions(digests)?
```

Comment:
Isn't it possible that some transactions will not be returned from this call at all?
I think in such case we should also return false right away.

Reply:
6270a5c594